### PR TITLE
Add experimental text colors

### PR DESCRIPTION
**Pull request by kyledurand**
First round of text color additions

From mapped values [here](https://www.figma.com/file/hL9XRyHGqU8wUtLdOgs0vI/Polaris-Uplift-Library?type=design&node-id=9011%3A154956&t=A1SKamPybQ0FYnpn-1)

Part of https://github.com/Shopify/polaris-summer-editions/issues/12

### CODE REVIEW

**Review by laurkim — APPROVED** [review]
Checked locally and the storybook for the AppProvider is failing because in the `WithSummerEditionsFeature` story, `CheckFeature` is trying to render AlphaCard which was renamed to Card. 

It looks like that is resolved in `v11.x.x` so probably just needs a rebase or merge off latest version of that branch 👍

File: polaris-tokens/src/token-groups/color.ts [hunk 1]
```diff
@@ -618,94 +618,117 @@ export const color: {
...
     description: '',
   },
   'color-text-success': {
     value: colors.green[700],
+    valueExperimental: colorsExperimental.green[15],
     description: '',
   },
   'color-text-primary': {
     value: colors.green[700],
+    valueExperimental: colorsExperimental.green[14],
```
Comment (laurkim) [inline polaris-tokens/src/token-groups/color.ts]: Just double checking but in the Figma, it looks like `text-primary` maps to `GR14` on the [Figma](https://www.figma.com/file/hL9XRyHGqU8wUtLdOgs0vI/Polaris-Uplift-Library?type=design&node-id=9011%3A154956&t=KuB4j77yg0H5sAso-1), not `G14`

File: polaris-tokens/src/token-groups/color.ts [hunk 2]
```diff
@@ -618,94 +618,117 @@ export const color: {
...
     description: '',
   },
   'color-text-primary-hover': {
     value: colors.green[800],
+    valueExperimental: colorsExperimental.green[14],
```
Comment (laurkim) [inline polaris-tokens/src/token-groups/color.ts]: Same as above re: `GR14`

File: polaris-tokens/src/token-groups/color.ts (hunk 1 above)
Comment (kyledurand) [reply polaris-tokens/src/token-groups/color.ts]: Thanks! Copilot was a pretty big help with all this but far from perfect

**Review by sarahill — APPROVED** [review]

**Review by bernardojoaogarcia — APPROVED** [review]

**Pull request merged**